dataclasses_sql/decorators.py

```diff
@@ -27,9 +81,10 @@ def get(cls, idvalue):
         kwargs = {k: v for k, v in row.items() if k != "id"}
         instance = cls(**kwargs)  # TODO: Figure out how to construct a cls from row
         instance._rowid = row["id"]
+        instance._fetching = False
         return instance
 
     extra = {"__post_init__": post_init, "save": save}
-    cls = type(cls.__name__, (), {**cls.__dict__, **extra})
+    cls = type(cls.__name__, (SQLModel,), {**cls.__dict__, **extra})
```

Comment:
We're injecting a base class here, but for some reason the `__post_init__` constructor is not getting called. It doesn't hurt anything right now, but eventually this needs to be fixed for expected initializations to happen.

Reply:
Needed to reorder decorators